streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -1195,13 +1195,8 @@ public boolean commitNeeded() {
                         commitNeeded = true;
                         entry.setValue(offset - 1);
                     }
-                } catch (final TimeoutException error) {
-                    // the `consumer.position()` call should never block, because we know that we did process data
-                    // for the requested partition and thus the consumer should have a valid local position
-                    // that it can return immediately
-
-                    // hence, a `TimeoutException` indicates a bug and thus we rethrow it as fatal `IllegalStateException`
-                    throw new IllegalStateException(error);
+                } catch (final TimeoutException swallow) {
+                    log.debug("Could not get consumer position for partition {} due to: {}", partition, swallow);
```

Comment:
This won't print the exception. It's using https://www.javadoc.io/static/org.slf4j/slf4j-api/1.7.30/org/slf4j/Logger.html#debug-java.lang.String-java.lang.Object-java.lang.Object- , but the intent is probably to use https://www.javadoc.io/static/org.slf4j/slf4j-api/1.7.30/org/slf4j/Logger.html#debug-java.lang.String-java.lang.Throwable- , which requires a fully constructed string as the first arg.

```suggestion
                    log.debug(String.format("Could not get consumer position for partition %d", partition), swallow);
```

Reply:
+1

Reply:
should there not be the same change inside this method:
private Long findOffset(final TopicPartition partition) {

Reply:
Good question.

I think both cases are different:

1. In `commitNeeded()`, we call `consumer.position()` if the flag `commitNeeded == false`, ie, we know that we did not process any record, and thus, we cannot be sure that the consumer has a locally cached valid position. (Note, that we introduce this code in AK 3.1.0 release, and it seems it was more or less copied from `findOffset` which is older.)

2. `findOffset` on the other hand, is only called for partitions which have the `commitNeeded` flag set as `true` (cf `committableOffsetsAndMetadata()` where we call `findOffset`). This code was already added in 2.6 release. The ticket is also reported only for `commitNeeded` case above, indicating that this older code is correct, while the bug was only introduced in 3.1 when the code might have be copied incorrectly.

Does this make sense? Or do I miss anything?

Reply:
Makes sense. Thanks for detailed explanation